tests/Xunit/J2N.Collections.Generic.List.Tests/List.Generic.Tests.IndexOf.cs

```diff
@@ -252,5 +252,292 @@ public void LastIndexOf_int_int_OutOfRangeExceptions(int count)
+            Assert.Equal(2, list2.IndexOf(nanDefault, 1));
+            Assert.Equal(2, list2.IndexOf(nanVariant1, 1));
+            Assert.Equal(2, list2.IndexOf(nanVariant2, 1));
+
+            Assert.Equal(2, list2.IndexOf(nanDefault, 2, 3));
+            Assert.Equal(2, list2.IndexOf(nanVariant1, 2, 3));
+            Assert.Equal(2, list2.IndexOf(nanVariant2, 2, 3));
+        }
+
+        /// <summary>
+        /// Tests IndexOf with double NaN values. In Java/J2N, NaN == NaN, unlike .NET's BCL.
+        /// </summary>
+        [Fact]
+        public void IndexOf_DoubleNaN()
+        {
+            if (typeof(T) != typeof(double))
+                return;
+
+            // Test with different NaN bit patterns
+            double nanDefault = double.NaN; // 0x7FF8000000000000
+            double nanVariant1 = BitConverter.Int64BitsToDouble(0x7FF8000000000001);
+            double nanVariant2 = BitConverter.Int64BitsToDouble(0xFFF8000000000001);
+
+            List<double> list = (List<double>)(object)new List<T> { (T)(object)1.0d, (T)(object)nanDefault, (T)(object)3.0d };
+
+            // NaN should be found at index 1 regardless of NaN bit pattern
+            Assert.Equal(1, list.IndexOf(nanDefault));
+            Assert.Equal(1, list.IndexOf(nanVariant1));
+            Assert.Equal(1, list.IndexOf(nanVariant2));
+
+            Assert.Equal(1, list.IndexOf(nanDefault, 0));
+            Assert.Equal(1, list.IndexOf(nanVariant1, 0));
+            Assert.Equal(1, list.IndexOf(nanVariant2, 0));
+
+            Assert.Equal(1, list.IndexOf(nanDefault, 0, 3));
+            Assert.Equal(1, list.IndexOf(nanVariant1, 0, 3));
+            Assert.Equal(1, list.IndexOf(nanVariant2, 0, 3));
+
+            // NaN at different positions with multiple NaN bit patterns
+            var list2 = new List<double> { nanVariant1, 1.0d, nanDefault, 3.0d, nanVariant2 };
+            Assert.Equal(0, list2.IndexOf(nanDefault));
+            Assert.Equal(0, list2.IndexOf(nanVariant1));
+            Assert.Equal(0, list2.IndexOf(nanVariant2));
+
+            Assert.Equal(2, list2.IndexOf(nanDefault, 1));
+            Assert.Equal(2, list2.IndexOf(nanVariant1, 1));
+            Assert.Equal(2, list2.IndexOf(nanVariant2, 1));
+
+            Assert.Equal(2, list2.IndexOf(nanDefault, 2, 3));
+            Assert.Equal(2, list2.IndexOf(nanVariant1, 2, 3));
+            Assert.Equal(2, list2.IndexOf(nanVariant2, 2, 3));
+        }
+
+        /// <summary>
+        /// Tests LastIndexOf with float NaN values. In Java/J2N, NaN == NaN, unlike .NET's BCL.
+        /// </summary>
+        [Fact]
+        public void LastIndexOf_FloatNaN()
+        {
+            if (typeof(T) != typeof(float))
+                return;
+
+            // Test with different NaN bit patterns
+            float nanDefault = float.NaN; // 0x7FC00000
+            float nanVariant1 = BitConverter.Int32BitsToSingle(0x7FC00001);
+            float nanVariant2 = BitConverter.Int32BitsToSingle(0xFFC00001);
```

Comment:
The tests are now failing due to the fact that `BitConverter` didn't have this overload on .NET Framework. Ideally, these would be `const float` values rather than converted on the fly so we don't have to rely on a conversion to happen.

But failing that, we should switch these to use `J2N.BitConversion` rather than `System.BitConverter`. Note that the rest of J2N doesn't use `BitConverter` at all for floating point conversions outside of the BCL tests, both to align with the JDK and to cover this missing overload.